server/plotting/plotter.py

```diff
@@ -5,69 +5,68 @@
 
 def plot_csv(file_path):
     data_frame = pd.read_csv(file_path)
+
     midway_point = int(len(data_frame) / 2)
     data_set_one = data_frame[0:midway_point]
     data_set_two = data_frame[midway_point:len(data_frame)]
-    datasets = [data_set_one, data_set_two]
-    for dataset in datasets:
-        plot_graph(dataset, 'Age', 'Age Breakdown of Applicants', 'Number of Applicants', 'Age', False)
-        plot_graph(dataset, 'Race', 'Racial Breakdown of Applicants', 'Number of Applicants', 'Race', True)
-        plot_gender(dataset)
+    data_sets = [data_set_one, data_set_two]
+
+    for data_set in data_sets:
+        plot_graph(data_set, 'Age', 'Age Breakdown of Applicants', 'Number of Applicants', 'Age', False)
+        plot_vs_interviewed(data_set, 'Gender')
+        plot_vs_interviewed(data_set, 'Race')
 
 
-def plot_graph(data_frame, column, title, ylabel, xlabel, shouldUseHist):
-    if shouldUseHist:
+def plot_graph(data_frame, column: str, title: str, y_label: str, x_label: str, should_use_hist: bool):
+    if should_use_hist:
         data_frame[column].hist()
     else:
         data_frame[column].plot()
     plt.title(title)
-    plt.ylabel(ylabel)
-    plt.xlabel(xlabel)
+    plt.ylabel(y_label)
+    plt.xlabel(x_label)
 
     plt.show()
 
 
-def plot_gender(data_frame):
-    males = 0
-    males_interviewed = 0
-    females = 0
-    females_interviewed = 0
-
-    for index, row in data_frame.iterrows():
-        if row['Gender'] == 'M':
-            males += 1
-            males_interviewed += row.Interviewed
-        elif row['Gender'] == 'F':
-            females += 1
-            females_interviewed += row.Interviewed
-
-    n_groups = 2
-    male_stats = (males, males_interviewed)
-    female_stats = (females, females_interviewed)
+def plot_vs_interviewed(data_frame, column: str):
+    unique_column_values = data_frame[column].unique()
 
     plt.subplots()
-    index = np.arange(n_groups)
-    bar_width = 0.35
+    index = np.arange(2)
+    bar_width = 0.15
     opacity = 0.8
 
-    plt.bar(index, male_stats, bar_width,
-            alpha=opacity,
-            color='b',
-            label='Male')
+    iteration = 0
+    for value in unique_column_values:
+        if not isinstance(value, str):
+            continue
+
+        data = get_info_for_row(data_frame, column, value)
+        plt.bar(index + (bar_width * iteration), data, bar_width,
```

Comment:
Could be worth assigning (index + (bar_width * iteration)) to it's own var "bar_starting_point" but that can be done in another ticket